imports/plugins/core/versions/server/migrations/59_drop_indexes.js

```diff
@@ -0,0 +1,125 @@
+      Orders.dropIndex("c2_items.$.product.ancestors");
+      Orders.dropIndex("c2_items.$.product.createdAt");
+      Orders.dropIndex("c2_items.$.product.handle");
+      Orders.dropIndex("c2_items.$.product.hashtags");
+      Orders.dropIndex("c2_items.$.product.isDeleted");
+      Orders.dropIndex("c2_items.$.product.isVisible");
+      Orders.dropIndex("c2_items.$.product.shopId");
+      Orders.dropIndex("c2_items.$.product.workflow.status");
+      Orders.dropIndex("c2_items.$.shopId");
+      Orders.dropIndex("c2_items.$.variants.isDeleted");
+      Orders.dropIndex("c2_items.$.variants.isVisible");
+      Orders.dropIndex("c2_items.$.variants.shopId");
+      Orders.dropIndex("c2_items.$.variants.workflow.status");
+      Orders.dropIndex("c2_items.$.workflow.status");
+      Orders.dropIndex("c2_sessionId");
+      Orders.dropIndex("c2_shipping.$.items.$.productId");
+      Orders.dropIndex("c2_shipping.$.items.$.shopId");
+      Orders.dropIndex("c2_shipping.$.items.$.variantId");
+      Orders.dropIndex("c2_shipping.$.items.$.workflow.status");
+      Orders.dropIndex("c2_shipping.$.workflow.status");
+
+      Packages.dropIndex("c2_layout.$.layout");
+      Packages.dropIndex("c2_layout.$.structure.adminControlsFooter");
+      Packages.dropIndex("c2_layout.$.structure.dashboardControls");
+      Packages.dropIndex("c2_layout.$.structure.dashboardHeader");
+      Packages.dropIndex("c2_layout.$.structure.dashboardHeaderControls");
+      Packages.dropIndex("c2_layout.$.structure.layoutFooter");
+      Packages.dropIndex("c2_layout.$.structure.layoutHeader");
+      Packages.dropIndex("c2_layout.$.structure.notFound");
+      Packages.dropIndex("c2_layout.$.structure.template");
+      Packages.dropIndex("c2_name");
+      Packages.dropIndex("c2_registry.$.name");
+      Packages.dropIndex("c2_registry.$.route");
+      Packages.dropIndex("c2_shopId");
+
+      Products.dropIndex("c2_isDeleted");
+      Products.dropIndex("c2_isVisible");
+
+      Shops.dropIndex("c2_active");
+      Shops.dropIndex("c2_layout.$.layout");
+      Shops.dropIndex("c2_layout.$.structure.adminControlsFooter");
+      Shops.dropIndex("c2_layout.$.structure.dashboardControls");
+      Shops.dropIndex("c2_layout.$.structure.dashboardHeader");
+      Shops.dropIndex("c2_layout.$.structure.dashboardHeaderControls");
+      Shops.dropIndex("c2_layout.$.structure.layoutFooter");
+      Shops.dropIndex("c2_layout.$.structure.layoutHeader");
+      Shops.dropIndex("c2_layout.$.structure.notFound");
+      Shops.dropIndex("c2_layout.$.structure.template");
+      Shops.dropIndex("c2_shopType");
+      Shops.dropIndex("c2_workflow.status");
+
+      Translations.dropIndex("c2_i18n");
+      Translations.dropIndex("c2_shopId");
+    } catch (error) {
+      // This may fail if the index doesn't exist, which is what we want anyway
+      Logger.warn(error, "Caught error from dropIndex calls in migration 59");
```

Comment:
Question, not a blocker: Can other error cases catch here? Would it be worth the effort to handle a specific "index doesn't exist" or is it OK to swallow all?

Reply:
Agree that would have been better but in this case any other error should be highly unlikely and it is still logged, so won’t be swallowed completely